### DISCUSSION TITLE
Is keycloak vulnerable to CVE-2022-21449

### DISCUSSION THREAD

**Discussion by fcovatti** [keycloak/keycloak#11540] [Q&A]
I was checking this post, https://neilmadden.blog/2022/04/19/psychic-signatures-in-java/ and it mentions that it can affect signed JWTs and OIDC id tokens, so I wonder if it has any affect on keycloak as it uses java.

**Comment by 0x53A** [comment]
I am neither a Keycloak dev nor a Java developer in general but a Keycloak user.

From what I know, the CVE is in the Java Runtime, so it depends on your deployment model.

* If you run Keycloak as a normal java app, just update the Java Runtime and you're probably good.

* If you're running the Keycloak Docker container, as we are, then I believe you/we are also good:  
From what I can see, the container uses Java-11. It seems the CVE only affects Java 15 to 18.

But I'd also really appreciate an "official" response.

**Reply by zsiciarz** [reply]
As per https://openjdk.java.net/groups/vulnerability/advisories/2022-04-19, the vulnerability also affects Java 11.0.14 or earlier.

**Reply by vanrar68** [reply]
The OpenJDK cumulative update addresses multiple CVEs. The one mentioned in this conversation (CVE-2022-21449, related to ECDSA signature) only affects OpenJDK 15+

**Reply by zsiciarz** [reply]
@vanrar68 the advisory you linked to says "Supported versions that are affected are Oracle Java SE: 7u331, 8u321, 11.0.14, 17.0.2, 18; Oracle GraalVM Enterprise Edition: 20.3.5, 21.3.1 and 22.0.0.2.". Although it doesn't say "or earlier" like the cumulative update does.

**Reply by vanrar68** [reply]
Actually I didn't link anything, just pasted the CVE string and Github picked it up and transformed it into a link to an internal advisory (related to Oracle products only).
The OpenJDK advisory clearly states that only OpenJDK 15+ is affected by this very specific CVE (other OpenJDK versions affected by other CVEs). So I guess it's a good idea to ensure your Keycloak instances are based on an up to date OpenJDK framework.

**Reply by 0x53A** [reply]
so, "no, OpenJDK11.x is not affected by CVE-2022-21449", BUT "yes, OpenJDK11.x is affected by OTHER CVEs, and so should be updated regardless"? Is that correct?

**Comment by abstractj** [comment]
Please read https://github.com/keycloak/keycloak/discussions/11754
